In [2]:
%load_ext autoreload
%autoreload 2

import helper

## DeepConvLSTM
### PAMAP2

In [ ]:
helper.train_LSTM_validation_model(
    helper.load_PAMAP2,
    total_activities = 7,
    val_pc = 0.3,
    batch_size = 32,
    data_size = (3, 9, 100),
    hidden_size = 256,
    lr = 0.0001,
    max_epochs = 20,
    tensorboard_save_dir = "Validation_LSTM_logs",
    tensorboard_name_prefix = "PAMAP2",
    monitor = "val_f1_score"
    )

### RWHAR

In [ ]:
helper.train_LSTM_validation_model(
    helper.load_RWHAR,
    total_activities = 8,
    val_pc = 0.3,
    batch_size = 64,
    data_size = (6, 50),
    hidden_size = 512,
    conv_filter = (3,5), 
    conv_padding = (1,2),
    lr = 0.0001,
    max_epochs = 100,
    tensorboard_save_dir = "Validation_LSTM_logs",
    tensorboard_name_prefix = "RWHAR",
    monitor = "val_f1_score"
    )

## Transformer
### PAMAP2

In [ ]:
helper.train_transformer_validation_model(
                helper.load_PAMAP2,
                total_activities = 7,
                val_pc = 0.3,
                batch_size = 32,
                data_size = (27, 100),
                nhead = 1,
                dim_feedforward = 2048,
                num_layer = 1,
                lr = 0.0001,
                max_epochs = 100,
                tensorboard_save_dir = "Validation_transformer_logs",
                tensorboard_name_prefix = "PAMAP2",
                monitor = "val_f1_score",)

### RWHAR

In [3]:
helper.train_transformer_validation_model(
                helper.load_RWHAR,
                total_activities = 8,
                val_pc = 0.3,
                batch_size = 64,
                data_size = (6, 50),
                nhead = 5,
                dim_feedforward = 10000,
                dropout = 0.3,
                num_layer = 5,
                lr = 0.0001,
                max_epochs = 100,
                tensorboard_save_dir = "Validation_transformer_logs",
                tensorboard_name_prefix = "RWHAR",
                monitor = "val_f1_score",)

File exists. Loading
Windowing
Done!
Train weights : 
 [[0.09102216]
 [0.11795568]
 [0.02286296]
 [0.14979647]
 [0.16858887]
 [0.15049751]
 [0.14907282]
 [0.15020353]] 
Validation weights : 
 [[0.09318769]
 [0.11133977]
 [0.02253179]
 [0.14959633]
 [0.16917313]
 [0.15001847]
 [0.1496491 ]
 [0.15450372]]


GPU available: True, used: True
TPU available: None, using: 0 TPU cores

  | Name      | Type                  | Params
----------------------------------------------------
0 | model     | TransformerClassifier | 5.1 M 
1 | criterion | CrossEntropyLoss      | 0     
----------------------------------------------------
5.1 M     Trainable params
0         Non-trainable params
5.1 M     Total params
20.572    Total estimated model params size (MB)


Validation sanity check: 0it [00:00, ?it/s]

Training: 0it [00:00, ?it/s]

Validating: 0it [00:00, ?it/s]

Validating: 0it [00:00, ?it/s]

Validating: 0it [00:00, ?it/s]

Validating: 0it [00:00, ?it/s]

Validating: 0it [00:00, ?it/s]

Validating: 0it [00:00, ?it/s]

Validating: 0it [00:00, ?it/s]

Validating: 0it [00:00, ?it/s]

Validating: 0it [00:00, ?it/s]

Validating: 0it [00:00, ?it/s]

Validating: 0it [00:00, ?it/s]

Validating: 0it [00:00, ?it/s]

Validating: 0it [00:00, ?it/s]

Validating: 0it [00:00, ?it/s]

Validating: 0it [00:00, ?it/s]

Validating: 0it [00:00, ?it/s]

Validating: 0it [00:00, ?it/s]

Validating: 0it [00:00, ?it/s]

Validating: 0it [00:00, ?it/s]

Validating: 0it [00:00, ?it/s]

Validating: 0it [00:00, ?it/s]

Validating: 0it [00:00, ?it/s]

Validating: 0it [00:00, ?it/s]

Validating: 0it [00:00, ?it/s]

Validating: 0it [00:00, ?it/s]

Validating: 0it [00:00, ?it/s]

Validating: 0it [00:00, ?it/s]

Validating: 0it [00:00, ?it/s]

Validating: 0it [00:00, ?it/s]

Validating: 0it [00:00, ?it/s]

Validating: 0it [00:00, ?it/s]

Validating: 0it [00:00, ?it/s]

Validating: 0it [00:00, ?it/s]

Validating: 0it [00:00, ?it/s]

Validating: 0it [00:00, ?it/s]

Validating: 0it [00:00, ?it/s]

Validating: 0it [00:00, ?it/s]

Validating: 0it [00:00, ?it/s]

Validating: 0it [00:00, ?it/s]

Validating: 0it [00:00, ?it/s]

Validating: 0it [00:00, ?it/s]

Validating: 0it [00:00, ?it/s]

Validating: 0it [00:00, ?it/s]

Validating: 0it [00:00, ?it/s]

Validating: 0it [00:00, ?it/s]

Validating: 0it [00:00, ?it/s]

Validating: 0it [00:00, ?it/s]

Validating: 0it [00:00, ?it/s]

Validating: 0it [00:00, ?it/s]

Validating: 0it [00:00, ?it/s]

Validating: 0it [00:00, ?it/s]

Validating: 0it [00:00, ?it/s]

Validating: 0it [00:00, ?it/s]

Validating: 0it [00:00, ?it/s]

Validating: 0it [00:00, ?it/s]

## Saved Model Score PAMAP2
Location : logs/PAMAP2_LSTM/version_2/checkpoints/epoch=8.ckpt

F1 Score : 96.21%

Loss     : 0.1168

## Parameters
LR           : 0.0001

Optimizer    : Adamax

Epochs       : 8

LR Scheduler : None

hidden_size : 256

## Saved Model Score
Location : logs/PAMAP2_transformer/version_11/checkpoints/epoch=49.ckpt

F1 Score : 96.96%

Loss     : 0.1113

## Parameters
LR           : 0.0001

Optimizer    : Adamax

Epochs       : 49

LR Scheduler : None

in_channels = 27, output_size = 7, d_model = 100, nhead = 1, dim_feedforward = 2048, 

## Saved model RWHAR

### Transformer Classifier

Version : 3

location : Validation_transformer_logs/RWHAR/version_3/checkpoints/epoch=66-step=14471.ckpt

F1 score : 88.7%

loss : 0.3502

in_channels = 6, output_size = 8, d_model = 50, nhead = 5, dim_feedforward = 10000, num_layers = 5

### LSTM model

version : 1

location : Validation_LSTM_logs/RWHAR/version_1/checkpoints/epoch=14-step=3239.ckpt

F1 score : 86.22%

loss : 0.3768

in_channels = 6,input_size = 50, hidden_size = 512, output_size = 8, conv_filter = (3,5), conv_padding = (1,2)
